### ISSUE TITLE
adaptererror connection already registered

### ISSUE THREAD

**Issue by traskat** [balderdashy/sails-mysql#252]
Hi,

i am getting this error.

```
DB ERROR:  { [AdapterError: Connection is already registered]
  name: 'AdapterError',
  message: 'Connection is already registered' }
```

If i do a one request to my app. it works. The Error appears only at the second request.
I am using waterline with the sails-mysql adapter. 

it is the latest version of waterline and sails-mysql.

thanks and best

**Comment by traskat** [comment]
Hi,

i investigated a some time.

For me it works till now. i didn't test it very much.
In the file. [register.js](https://github.com/balderdashy/sails-mysql/blob/master/lib/connections/register.js)

i commented out this line. So it looks like that.

```
return function registerConnection (connection, collections, cb) {

    // Validate arguments
    if(!connection.identity) return cb(Errors.IdentityMissing);
    //if(connections[connection.identity]) return cb(Errors.IdentityDuplicate);

    // Build up a schema for this connection that can be used throughout the adapter
    var schema = {};

    _.each(_.keys(collections), function(coll) {
      var collection = collections[coll];
      if(!collection) return;
```

If someone could test it.

thanks

**Comment by particlebanana** [comment]
Now sure why `registerConnection` would be called more than once. Are you using clusters or anything?

**Comment by sailsbot** [comment]
Thanks for posting, @traskat.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!

**Comment by devinivy** [comment]
This is a bug– reopening.

**Comment by particlebanana** [comment]
@traskat is there anyway you can create a reliable example that reproduces this issue and link to it here?

**Comment by traskat** [comment]
Hi,

@particlebanana to your first question i am not using clusters or something like that.

config looks like that:

```
{
"adapters": {
		"default": "mysql",
		"mysql": "mysql-sails"
	},
	"connections":{
		"mysql-remote": {
			"module"			: "mysql-sails",
			"adapter"			: "mysql",
			"host"				: "db.example.com",
			"port"				: "3306",
			"user"				: "mrroot",
			"password"			: "password",
			"database"			: "my_db",
			"debug"				: false,
			"pool"				: true,
			"connectionLimit"	: 10,
			"waitForConnections": false
		}
   }
}
```
here a model example:
```
module.exports = {
	tablename: 'users',
	connection: 'mysql-remote',
	identity: 'users',
	migrate: 'alter',
	attributes: {
		id: {
			type: 'integer',
			autoIncrement: true,
			unique: true,
			index: true,
			primaryKey: true
		},
		username: {
			type: 'string',
			required: true
		},
		email: {
			type: 'string',
			required: true
		},
		password: {
			type: 'string',
			required: true
		}
	},
```

little example:
```
var Waterline = require('waterline');
var sailsMysql = require('sails-mysql');
/**
		initiaialize waterline orm
	**/
	var connection = new Waterline();

	var usersMod = require('./models/user');

	usersMod.connection = 'mysql-remote';
	config.adapters.mysql = sailsMysql;

	var users = Waterline.Collection.extend(usersMod);

	connection.loadCollection(users);

       connection.initialize(config, function(err, models) {
			if (err) {
				console.log('DB ERROR: ', err);
				return;
			}
			
			req.models = models.collections;
			req.connections = models.connections;
			next();
		});
```

Hope that is enough. If not please give some feedback and i will try to make a link accessable to problem repository.
thanks normen

**Comment by traskat** [comment]
also i found the same issue here
balderdashy/sails#2332

maybe it is a adapter based problem?

**Comment by particlebanana** [comment]
@traskat ahh that helps. Looks like the patch made [here](https://github.com/balderdashy/sails-postgresql/pull/127/files) needs to be applied in this library as well.

**Comment by sgress454** [comment]
Applied that patch in d2d38ad
